lib/ui/window/platform_configuration.cc

```diff
@@ -377,7 +377,18 @@ void PlatformConfiguration::BeginFrame(fml::TimePoint frameTime,
   }
   tonic::DartState::Scope scope(dart_state);
 
-  int64_t microseconds = (frameTime - fml::TimePoint()).ToMicroseconds();
+  int64_t microseconds = frameTime.ToEpochDelta().ToMicroseconds();
+
+  static int64_t last_microseconds = 0;
+  if (last_microseconds > microseconds) {
+    // Do not allow time traveling frametimes
+    // github.com/flutter/flutter/issues/106277
+    FML_LOG(ERROR)
+        << "Reported frame time is older than the last one; clamping. "
+        << microseconds << " < " << last_microseconds
+        << " ~= " << last_microseconds - microseconds;
+    microseconds = last_microseconds;
```

Comment:
Oh! Almost missed this. Could this be why the repro wasn't working?
```suggestion
    last_microseconds = microseconds;
```

Reply:
nvm, this is the bigger issue: https://github.com/flutter/engine/pull/55310#discussion_r1767837151

Reply:
I was testing before this and the repo wasn't caching. This is just the "patch" to make life better.